client/blocks/login/login-form.jsx

```diff
@@ -231,6 +243,35 @@ export class LoginForm extends Component {
 		this.usernameOrEmail = input;
 	};
 
+	jetpackCreateAccountWithMagicLink() {
+		// When a user enters a username or an email address that doesn't have a corresponding
+		// WPCOM account, we need to figure out whether the user entered a username or an email
+		// address. If the user entered an email address, we can safely attempt to create a WPCOM
+		// account for this user with the help of magic links. On the other hand, if the user entered
+		// a username, we need to prompt the user specifically for an email address to proceed with
+		// the WPCOM account creation with magic links.
+
+		const isEmailAddress = includes( this.state.usernameOrEmail, '@' );
```

Comment:
I wonder if we should better validate that it is, in fact, a full email address here? Someone could enter a username containing an @ symbol in it, (like a twitter username), but it's not an email address.

Reply:
That's a good question. I copy that check from another file involved in the login flow, so I assumed it would work fine. 

The good thing is that I just validated that an @ symbol can't be used for usernames. So, I guess we are safe. Thanks for keeping a close eye on everything.